src/dgipy/vcf.py

```diff
@@ -0,0 +1,202 @@
+"""Provides methods for annotating VCF with DGIdb data"""
+
+import contextlib
+from itertools import groupby
+
+import pandas as pd
+import pysam
+import requests
+from tqdm import tqdm
+
+import dgipy
+
+# Sample usage: import vcf
+#               data = vcf.annotate('link/to/file',chr='chr#')
+
+
+# TODO: Probably need another class as a wrapper object rather than putting it all in a list
+# Class would have analogous display methods but also allow access to individual GeneResults
+class GeneResult:
+    """A gene result from original VCF
+
+    Attributes
+    ----------
+    gene : str
+        The name of the gene
+    records : list
+        The mapped records from original VCF
+    interactions : pd.DataFrame
+        The interactions obtained from DGIdb
+
+    Methods
+    -------
+    search_interactions(gene: str) -> pd.DataFrame
+        Search drug-gene interactions for given gene in DGIdb
+
+    """
+
+    def __init__(self, data: list) -> None:
+        """Initialize a gene with a name, VCF records, and drug-gene interactions"""
+        # TODO: handle genes without names, 'novel transript'
+        try:
+            self.gene = data[0]["name"]
+        except:
+            self.gene = "None"
+
+        self.records = data
+        self.interactions = self.search_interactions(self.gene)
+
+        # TODO: handle app searches with blank lists [], currently FDA resource hangs for awhile?
+        if not list(self.interactions["drug"].values):
+            self.applications = "None"
+        else:
+            self.applications = self.search_applications(
+                list(self.interactions["drug"].values)
+            )
+
+        self.gene_info = self.grab_gene_info(self.gene)
+        self.categories = self.grab_categories(self.gene)
+
+    # TODO: These can probably just be simpilifed to direct assignment during __init__ but might be useful this way too
```

Comment:
Yeah, if we're not adding any additional transforms or functions while calling these methods, we might not need a second level of function call to wrap them

Reply:
leaving these in for now just in case we think of additional ways to transform.